DESIGN-PROPOSALS.md

```diff
@@ -0,0 +1,82 @@
+# Design Proposal Template
+
+**Authors**: alice alice@example.com, bob jones [bob@example.com](mailto:bob@example.com)
+
+**Begin Design Discussion**: YYYY-MM-DD
+
+**(optional) Status: **implementable, accepted, deferred, rejected, withdrawn, or replaced
```

Comment:
```suggestion
**(optional) Status: **implementable, accepted, deferred, rejected, withdrawn, or replaced

**Checklist**:

- [ ] Prerequisite 1
- [ ] Prerequisite 2
- [ ] Component 1
- [ ] Component 2
- [ ] Docs
- [ ] Tests


```

We should have a checklist in here so that people can track how much of the proposal has been implemented.  It's part of the KEP template.